File: JPR/src/controllers/analizador/tablaSimbolos/Simbolo.ts
import Tipo from "./Tipo";

export default class Simbolo
{
    private tipo: Tipo;
    private identificador: String;
    private valor: any;

    constructor(tipo: Tipo, identificador: String, valor?:any)
    {
        this.tipo = tipo;
        this.identificador = identificador;
        if(valor)
        {
            this.valor = valor;
        }
        else
        {
            this.valor = null;
        }

    }

    public getIdentificador() {
        return this.identificador;
    }

    public setIdentificador(identificador: String) {
        this.identificador = identificador;
    }
    public getTipo(){
        return this.tipo;
    }
    public setTipo(tipo: Tipo)
    {
        this.tipo = tipo;
    }

    public getValor(){
        return this.valor;
    }
    public setValor(valor: String)
    {
        this.valor = valor;
    }
}

File: JPRJava/JPR/src/TablaSimbolos/Simbolo.java
package TablaSimbolos;

/**
 *
 * @author Pavel
 */
public class Simbolo {

    private Tipo tipo;
    private String identificador;
    private Object valor;


    public Simbolo(Tipo tipo, String identificador) {
        this.tipo = tipo;
        this.identificador = identificador;
    }

    public Simbolo(Tipo tipo, String identificador, Object valor) {
        this.tipo = tipo;
        this.identificador = identificador;
        this.valor = valor;
    }

    public String getIdentificador() {
        return identificador;
    }

    public void setIdentificador(String identificador) {
        this.identificador = identificador;
    }

    public Tipo getTipo() {
        return tipo;
    }

    public void setTipo(Tipo tipo) {
        this.tipo = tipo;
    }

    public Object getValor() {
        return valor;
    }

    public void setValor(Object valor) {
        this.valor = valor;
    }
}
